app/src/main/java/com/zeapo/pwdstore/git/BaseGitActivity.kt

```diff
@@ -116,13 +115,8 @@ abstract class BaseGitActivity : AppCompatActivity() {
                 result
             }
         }
-        if (hostname != previousHostname) {
-            encryptedSettings.edit().apply {
-                when (protocol) {
-                    Protocol.Ssh -> remove("ssh_key_passphrase")
-                    Protocol.Https -> remove("https_password")
-                }
-            }.apply()
+        if (hostname != previousHostname && protocol == Protocol.Https) {
+            encryptedSettings.edit().remove("https_password").apply()
         }
         if (!valid)
```

Comment:
This seems like a bug, doesn't it? I'm currently rewriting this function from scratch though, so don't bother.